### DISCUSSION TITLE
Conditional onBeforeRender

### DISCUSSION THREAD

**Discussion by brillout** [vikejs/vike#676] [Discussion]
Moved from Discord:

> Hi, I am using react-query with onBeforeRender to get detail of book from API during SSR. The thing is that I want this to happen only if user visits directly the detail page, but if the user is coming from navigate or a I want it to skip the onBeforeRender . Do you have any suggestions on how to achieve that? I thought that maybe doNotPrerender but I have no clue what the condition would be.

**Comment by kruzliak-juraj** [comment]
Thanks, sorry for posting it there.

**Comment by brillout** [comment]
I don't understand your use case, but I'd assume that implementing the condition as an `if` branch *inside* `onBeforeRender()` should do the trick.

Also note that the V1 design will allow conditional hooks, but I strongly suspect you don't really need that.
